src/frontend/components/SchedulingFields/index.tsx

```diff
@@ -0,0 +1,224 @@
+    id: 'component.SchedulingFields.startingAtTextLabel',
+  },
+  estimatedDurationTextLabel: {
+    defaultMessage: 'Estimated duration',
+    description:
+      'Label of the input field to add/update the estimated duration of the scheduled event',
+    id: 'component.SchedulingFields.estimatedDurationTextLabel',
+  },
+});
+
+// build a list of times from 09h00 to 19h30
+const currentTimeForSuggestions = Duration.fromObject({ hours: 9 });
+const allTimesSuggestions = Array.from({ length: 22 }, (_v, k) => k * 30).map(
+  (minutes) => currentTimeForSuggestions.plus({ minutes }).toFormat('hh:mm'),
+);
+
+const displayedDuration = (duration: Duration) => {
+  return duration.toFormat('h:mm');
+};
+
+const returnedDuration = (durationString: string) => {
+  if (durationString.length === 4) durationString = '0' + durationString;
+  return Duration.fromISOTime(durationString).toFormat('hh:mm:ss');
+};
+
+// build a list of durations from 15 minutes to 5 hours
+const currentDurationForSuggestions = Duration.fromObject({ minutes: 15 });
+const allDurationSuggestions = Array.from(
+  { length: 20 },
+  (_v, k) => k * 15,
+).map((minutes) =>
+  displayedDuration(currentDurationForSuggestions.plus({ minutes })),
+);
+
+interface SchedulingFieldsProps {
+  startingAt: Nullable<string>;
+  estimatedDuration: Nullable<string>;
+  onStartingAtChange?: (date: string) => void;
+  onEstimatedDurationChange?: (duration: string) => void;
+  margin?: MarginType;
+}
+
+export const SchedulingFields = ({
+  startingAt,
+  estimatedDuration,
+  onStartingAtChange = () => {},
+  onEstimatedDurationChange,
+  margin,
+}: SchedulingFieldsProps) => {
+  const intl = useIntl();
+  Settings.defaultLocale = intl.locale;
+  Settings.defaultZone = 'utc';
+
+  const [currentStartingAtDate, setCurrentStartingAtDate] = React.useState(
+    startingAt
+      ? DateTime.fromISO(startingAt)
+          .set({ hour: 0, minute: 0, second: 0 })
+          .toISO()
+      : '',
+  );
+  const [timeSuggestions, setTimeSuggestions] =
+    React.useState(allTimesSuggestions);
+  const [currentStartingAtTime, setCurrentStartingAtTime] = React.useState(
+    startingAt
+      ? DateTime.fromISO(startingAt).toLocaleString(DateTime.TIME_24_SIMPLE)
+      : '',
+  );
+  const [durationSuggestions, setDurationSuggestions] = React.useState(
+    allDurationSuggestions,
+  );
+  const [currentEstimatedDuration, setCurrentEstimatedDuration] =
+    React.useState(
+      estimatedDuration
+        ? displayedDuration(Duration.fromISOTime(estimatedDuration))
+        : '',
+    );
+
+  const onStartingAtDateInputChange = (event: { value: string | string[] }) => {
+    if (Array.isArray(event.value)) {
+      if (event.value.length > 0) {
+        setCurrentStartingAtDate(event.value[0]);
+      }
+    } else {
+      setCurrentStartingAtDate(event.value);
+    }
+    const updatedStartingAt = DateTime.fromISO(event.value as string).plus(
```

Comment:
Is there any reason not to  handle this aswell ?

Reply:
I just forgot…